### DISCUSSION TITLE
Function on close of SelectMenu modal

### DISCUSSION THREAD

**Discussion by maskott** [segmentio/evergreen#1320] [Q&A] [answered]
I have seen the [description of `SelectMenuContent` in the docs](https://evergreen.segment.com/components/select-menu/props), but I do not understand if/how I can use the `close` attribute. 

Is there a way for me to trigger a custom function upon the close of a `SelectMenu`'s selection modal?

**Comment by brandongregoryscott** [comment, marked as answer]
Hi @maskott! I believe the `onClose` prop is what you're looking for. It's not obvious from the docs, but `SelectMenuProps` extends most of the `PopoverProps`, including `onClose` (the TS definitions should reflect this!): 

```ts
export interface SelectMenuProps extends Omit<PopoverProps, 'position' | 'content'> {
  /**
   * The title of the Select Menu.
   */
  title?: string
  // ... more props
  
export interface PopoverProps {
  // ... more props
  onOpen?: () => void
  onClose?: () => void
}
```

Hopefully this helps!
